GitUtil.go

```diff
@@ -83,7 +93,7 @@ func CloneAndCheckout(ciProjectDetails []CiProjectDetails) error {
 			log.Println("checkout commit in webhook : ", targetCheckout)
 
 			// checkout target hash
-			_, msgMsg, cErr = gitCli.Checkout(filepath.Join(workingDir, prj.CheckoutPath), targetCheckout)
+			msgMsg, cErr = Checkout(gitCli, prj.CheckoutPath, targetCheckout, authMode, prj.FetchSubmodules, auth.Username, auth.Password, prj.GitRepository)
```

Comment:
we always fetch all submodules? is it possible to fetch specific submodule? is that a usecase?

Reply:
Yes. We always fetch all the submodules. 
Fetching specific submodule is not the use case as of now.

Reply:
ok